dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.42/V2_42_26__Add_not_null_constraint_to_trackedentity.sql

```diff
@@ -0,0 +1,40 @@
+DO $$
+DECLARE
+results RECORD;
+inconsistent_records_count INTEGER;
+BEGIN
+
+SELECT COUNT(*) INTO inconsistent_records_count FROM trackedentity WHERE trackedentitytypeid IS NULL;
+
+IF inconsistent_records_count > 0 THEN
+        RAISE NOTICE 'Inconsistencies found: %', inconsistent_records_count;
+
+    FOR results IN
+    SELECT programid, trackedentityid
```

Comment:
Suggested changes
1. You can fetch the trackedEntityTypeId directly in this query by joining with program table so that it is already present in results
2. Include another where clause so that you also check TrackedentityTypeid is not null (from the enrollment table)
3. In the update section, you can directly use results.trackedEntityTypeId

What do you think?

Reply:
used set based operation and removed looping logic.